File: app/src/main/java/com/example/kinginstaller/StreamLogs.java
package com.example.kinginstaller;

public class StreamLogs {
    private String inputStreamLog;
    private String errorStreamLog;
    private String outputStreamLog;

    public StreamLogs() {}

    public String getInputStreamLog() {
        if (inputStreamLog == null) { return ""; }
        return inputStreamLog.trim();
    }

    public String getErrorStreamLog() {
        if (errorStreamLog == null) { return ""; }
        return errorStreamLog.trim();
    }

    public String getOutputStreamLog() {
        if (outputStreamLog == null) { return ""; }
        return outputStreamLog.trim();
    }

    public void setInputStreamLog(String inputStreamLog) { this.inputStreamLog = inputStreamLog; }

    public void setErrorStreamLog(String errorStreamLog) { this.errorStreamLog = errorStreamLog; }

    public void setOutputStreamLog(String outputStreamLog) {
        this.outputStreamLog = outputStreamLog;
    }

    public String getInputStreamLogWithLabel() {
        return "\tInputStream:\n\t\t" +
                getInputStreamLog().replaceAll("\n", "\n\t\t");
    }

    public String getErrorStreamLogWithLabel() {
        return "\tErrorStream:\n\t\t" +
                getErrorStreamLog().replaceAll("\n", "\n\t\t");
    }

    public String getOutputStreamLogWithLabel() {
        return "\tOutputStream:\n\t\t" +
                getOutputStreamLog().replaceAll("\n", "\n\t\t");
    }

    public String getStreamLogsWithLabels() {
        String result = "\n" + getOutputStreamLogWithLabel();
        if (!getInputStreamLog().isEmpty()) { result += "\n" + getInputStreamLogWithLabel(); }
        if (!getErrorStreamLog().isEmpty()) { result += "\n" + getErrorStreamLogWithLabel(); }
        return result;
    }
}

File: app/src/main/java/com/example/kinginstaller/StreamLogs.kt
package com.example.kinginstaller

class StreamLogs {
    private var inputStreamLog: String? = null
    private var errorStreamLog: String? = null
    private var outputStreamLog: String? = null

    fun getInputStreamLog(): String {
        if (inputStreamLog == null) {
            return ""
        }
        return inputStreamLog!!.trim { it <= ' ' }
    }

    fun getErrorStreamLog(): String {
        if (errorStreamLog == null) {
            return ""
        }
        return errorStreamLog!!.trim { it <= ' ' }
    }

    fun getOutputStreamLog(): String {
        if (outputStreamLog == null) {
            return ""
        }
        return outputStreamLog!!.trim { it <= ' ' }
    }

    fun setInputStreamLog(inputStreamLog: String?) {
        this.inputStreamLog = inputStreamLog
    }

    fun setErrorStreamLog(errorStreamLog: String?) {
        this.errorStreamLog = errorStreamLog
    }

    fun setOutputStreamLog(outputStreamLog: String?) {
        this.outputStreamLog = outputStreamLog
    }

    val inputStreamLogWithLabel: String
        get() = "\tInputStream:\n\t\t" +
                getInputStreamLog().replace("\n".toRegex(), "\n\t\t")

    val errorStreamLogWithLabel: String
        get() = "\tErrorStream:\n\t\t" +
                getErrorStreamLog().replace("\n".toRegex(), "\n\t\t")

    val outputStreamLogWithLabel: String
        get() = "\tOutputStream:\n\t\t" +
                getOutputStreamLog().replace("\n".toRegex(), "\n\t\t")

    val streamLogsWithLabels: String
        get() {
            var result = "\n" + this.outputStreamLogWithLabel
            if (!getInputStreamLog().isEmpty()) {
                result += "\n" + this.inputStreamLogWithLabel
            }
            if (!getErrorStreamLog().isEmpty()) {
                result += "\n" + this.errorStreamLogWithLabel
            }
            return result
        }
}
